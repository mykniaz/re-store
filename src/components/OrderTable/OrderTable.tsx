import * as React from 'react';

import OrderTableItem from './OrderTableItem';

export interface IOrderTableItem {
  id: number;
  title: string;
  count: number;
  total: number;
}

interface IProps {
  items: IOrderTableItem[];
  onIncrease: (id: number) => void;
  onDecrease: (id: number) => void;
  onDelete: (id: number) => void;
}

export const OrderTable: React.SFC<IProps> = ({ items, onIncrease, onDecrease, onDelete }) => {
  const tableRows = items.map((col, colIndex: number) => {
    return (
      <OrderTableItem
        key={col.id}
        col={col}
        colIndex={colIndex}
        onIncrease={onIncrease}
        onDecrease={onDecrease}
        onDelete={onDelete}
      />
    );
  });

  return (
    <table className="table order-table">
      <thead>
        <tr>
          <th scope="col">#</th>
          <th scope="col">Item</th>
          <th scope="col">Count</th>
          <th scope="col">Price</th>
          <th scope="col">Action</th>
        </tr>
      </thead>
      <tbody>
        {tableRows}
      </tbody>
    </table>
  );
};

export default OrderTable;
