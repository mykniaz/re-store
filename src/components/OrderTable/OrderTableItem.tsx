import * as React from 'react';
import { IOrderTableItem } from './OrderTable';

interface IProps {
  col: IOrderTableItem;
  colIndex: number;
  onIncrease: (id: number) => void;
  onDecrease: (id: number) => void;
  onDelete: (id: number) => void;
}

const OrderTableItem: React.SFC<IProps> = ({ col, colIndex, onDecrease, onIncrease, onDelete }) => {
  return (
    <tr>
      <th scope="row">{colIndex + 1}</th>
      <td>{col.title}</td>
      <td>{col.count}</td>
      <td>{col.total}$</td>
      <td>
        <div className="btn-group">
          <button
            className="btn btn-warning"
            onClick={() => onDecrease(col.id)}
          >
            -
          </button>
          <button
            className="btn btn-primary"
            onClick={() => onIncrease(col.id)}
          >
            +
          </button>
          <button
            className="btn btn-danger"
            onClick={() => onDelete(col.id)}
          >
            Remove
          </button>
        </div>
      </td>
    </tr>
  );
};

export default OrderTableItem;
