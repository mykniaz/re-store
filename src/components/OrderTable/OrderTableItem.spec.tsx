import * as React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';
import OrderTableItem from './OrderTableItem';
import { IOrderTableItem } from './OrderTable';

afterEach(cleanup);

const getTestItem = (id: number, title = 'Test Title', count = 2, total = 50): IOrderTableItem => ({
  id,
  title,
  count,
  total,
});

describe('Order', () => {
  const testProps = {
    col: getTestItem(1),
    colIndex: 0,
    onIncrease: () => {},
    onDecrease: () => {},
    onDelete: () => {},
  };

  describe('Order Item Render', () => {
    const nextProps = {
      ...testProps,
    };

    const OrderItemContainer = shallow(<OrderTableItem {...nextProps} />);

    it('isRenderProperly', () => {
      expect(OrderItemContainer).toMatchSnapshot();
    });
  });

  describe('Order Click', () => {
    const nextProps = {
      ...testProps,
      onIncrease: jest.fn(),
      onDecrease: jest.fn(),
      onDelete: jest.fn(),
    };

    const OrderItemContainer = shallow(<OrderTableItem {...nextProps} />);

    const onIncreaseButton = OrderItemContainer.find('button.btn-primary');
    const onDecreaseButton = OrderItemContainer.find('button.btn-warning');
    const onDeleteButton = OrderItemContainer.find('button.btn-danger');

    it('onIncreaseEvent', () => {
      onIncreaseButton.simulate('click');

      expect(nextProps.onIncrease).toBeCalled();
    });

    it('onDecreaseEvent', () => {
      onDecreaseButton.simulate('click');

      expect(nextProps.onIncrease).toBeCalled();
    });

    it('onDeleteEvent', () => {
      onDeleteButton.simulate('click');

      expect(nextProps.onIncrease).toBeCalled();
    });
  });
});
