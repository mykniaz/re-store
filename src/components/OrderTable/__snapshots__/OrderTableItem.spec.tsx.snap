// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Order Item Render isRenderProperly 1`] = `ShallowWrapper {}`;
